editor/modes/visual-editor/block.js

```diff
@@ -270,7 +313,81 @@ class VisualEditorBlock extends Component {
 
 	onKeyUp( event ) {
 		this.removeOrDeselect( event );
-		this.handleArrowKey( event );
+		//this.handleArrowKey( event );
+	}
+
+	handleToolbarTabCycle( event ) {
+		const { keyCode, shiftKey, target } = event;
+		if ( keyCode !== TAB || ! this.node ) {
+			return;
+		}
+		const block = findDOMNode( this.node );
+		const isVisible = ( elem ) => elem && ! ( elem.offsetWidth <= 0 || elem.offsetHeight <= 0 );
+		const allToolbars = filter( Array.from( block.querySelectorAll( '.components-toolbar' ) ), isVisible );
+		const settingsMenu = filter( [ block.querySelector( '.editor-block-settings-menu' ) ], isVisible );
+		const moverMenu = filter( [ block.querySelector( '.editor-block-mover' ) ], isVisible );
+		const allCycle = [ ...allToolbars, ...settingsMenu, ...moverMenu ];
+
+		if ( shiftKey ) {
+			allCycle.reverse();
+		}
+
+		const targetIndex = findIndex( allCycle, ( node ) => node.contains( target ) );
+
+		if ( targetIndex === -1 ) {
+			return;
+		}
+
+		// we have a toolbar selected so we should be tabbing between toolbars
+		event.preventDefault();
+		event.stopPropagation();
+
+		if ( allCycle.length > 1 ) {
+			const nextToolbar = ( targetIndex + 1 >= allCycle.length ) ? allCycle[ 0 ] : allCycle[ targetIndex + 1 ];
+
+			const firstFocusableItem = nextToolbar.querySelector( 'button, *[tabindex]' );
```

Comment:
This isn't fully representative of finding first focusable, i.e. would match a disabled button, or `tabIndex=-1`.

Related effort ongoing in #2323